transaction/types/src/block_version.rs

```diff
@@ -13,9 +13,22 @@ use serde::{Deserialize, Serialize};
 /// the network is not supported, generally by requesting users to upgrade their
 /// software.
 ///
-/// If you need to manipulate block versions that cannot be understood by your
-/// version of `mc-transaction-core`, then you should use u32 to represent
-/// block version numbers.
+/// If you need to manipulate block versions that come from the network, you
+/// should use u32 to represent that.
+///
+/// Then, if you need to e.g. sign a transaction,
+/// you should try to convert them to BlockVersion.
+/// If that conversion fails, it means that this set of rules
+/// is not understood by your version of `mc-transaction-core`.
+/// This means that your build has reached end-of-life, and the user needs to
+/// update.
```

Comment:
Nit: Mixing "you" and "the user"
```suggestion
/// This means that your build has reached end-of-life, and needs an update.
```